data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/PipelineTests.java

```diff
@@ -42,6 +43,7 @@ class PipelineTests {
     private static final int TEST_READ_BATCH_TIMEOUT = 3000;
     private static final int TEST_PROCESSOR_THREADS = 1;
     private static final String TEST_PIPELINE_NAME = "test-pipeline";
+    private static final long PROCESSOR_SHUTDOWN_TIMEOUT = new Random().nextLong();
```

Comment:
I'd recommend that you make this a non-final, non-static field and set it up in an `@BeforeEach` method. Also, I have the same question about negative numbers.